pkg/expression/infer_pushdown.go

```diff
@@ -75,9 +89,24 @@ func canFuncBePushed(ctx EvalContext, sf *ScalarFunction, storeType kv.StoreType
 	}
 
 	if ret {
-		funcFullName := fmt.Sprintf("%s.%s", sf.FuncName.L, strings.ToLower(sf.Function.PbCode().String()))
+		if result := IsPushDownEnabled(sf.FuncName.L, storeType); !result {
+			return result
+		}
+		// if DefaultExprPushDownBlacklist is nil, it means that the push down blacklist is not initialized yet.
+		// so we return true to allow the push down.
+		// but `storeType != kv.TiFlash && name == ast.AggFuncApproxCountDistinct`
+		// is a special case, we should not allow the push down.
+		// next we will check the funcFullName which contains `.` , this special case will not match the funcFullName.
+		// If the DefaultExprPushDownBlacklist is nil, we can return true directly.
+		DefaultExprPushDownBlacklistMap := DefaultExprPushDownBlacklist.Load()
```

Comment:
also defaultExprPushDownBlacklistMap should be OK

Reply:
Updated.